components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserEmailVerificationHandler.java

```diff
@@ -507,4 +525,46 @@ private boolean isVerifyEmailClaimAvailable(Map<String, String> claims) {
         return (claims.containsKey(IdentityRecoveryConstants.VERIFY_EMAIL_CLIAM) &&
                 Boolean.parseBoolean(claims.get(IdentityRecoveryConstants.VERIFY_EMAIL_CLIAM)));
     }
+
+    private void publishPreVerifyEmailClaimEvent(Map<String, Object> eventProperties)
+            throws IdentityEventException {
+
+        Map<String, Object> verifyEmailEventProperties = Utils.cloneMap(eventProperties);
+        Event identityMgtEvent = new Event(IdentityEventConstants.Event.PRE_VERIFY_EMAIL_CLAIM,
+                verifyEmailEventProperties);
+        IdentityRecoveryServiceDataHolder.getInstance().getIdentityEventService().handleEvent(identityMgtEvent);
+    }
+
+    private void publishPostVerifyEmailClaimEvent(Map<String, Object> eventProperties, String confirmationCode)
+            throws IdentityEventException {
+
+        Map<String, Object> verifyEmailEventProperties = Utils.cloneMap(eventProperties);
+        if (StringUtils.isNotBlank(confirmationCode)) {
+            verifyEmailEventProperties.put(IdentityRecoveryConstants.CONFIRMATION_CODE, confirmationCode);
+        }
+        Event identityMgtEvent = new Event(IdentityEventConstants.Event.POST_VERIFY_EMAIL_CLAIM,
+                verifyEmailEventProperties);
+        IdentityRecoveryServiceDataHolder.getInstance().getIdentityEventService().handleEvent(identityMgtEvent);
+    }
+
+    private void publishPreAddUserWithAskPasswordEvent(Map<String, Object> eventProperties)
+            throws IdentityEventException {
+
+        Map<String, Object> askPasswordEventProperties = Utils.cloneMap(eventProperties);
+        Event identityMgtEvent = new Event(IdentityEventConstants.Event.PRE_ADD_USER_WITH_ASK_PASSWORD,
+                askPasswordEventProperties);
+        IdentityRecoveryServiceDataHolder.getInstance().getIdentityEventService().handleEvent(identityMgtEvent);
+    }
+
+    private void publishPostAddUserWithAskPasswordEvent(Map<String, Object> eventProperties,
```

Comment:
Can't we have all this implementation in a single `event publish` method by passing the event name and the confirmation code as null when required

Reply:
Fixed in 61d7548fc71aca19ecb53cb10928da2bb00294d6